core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2431,6 +2431,26 @@ object Stream {
   def raiseError[F[x] >: Pure[x], O](e: Throwable): Stream[F, O] =
     fromFreeC(Algebra.raiseError(e))
 
+  /**
+    * Creates a random stream of integers using a random seed.
+    */
+  def random[F[_]](implicit F: Sync[F]): Stream[F, Int] =
+    Stream.eval(F.delay(new scala.util.Random())).flatMap { r =>
+      def loop: Stream[F, Int] = Stream.emit(r.nextInt) ++ loop
```

Comment:
Yeah, I used the same unfolding strategy as the `randomSeeded` implementation though in this case, we could use `F.delay`. Since `r` isn't observable to anyone outside this function, this scheme is safe though.

Reply:
I'd rather we used `repeatEval` or, if there are significant advantages to using `emit` in this case, add a small comment explaining why this is safe

Reply:
I'm fine with using `repeatEval` in this case but note in `randomSeeded` we need the emit trick to avoid a `Sync` dependency.

Reply:
I'm fine with limited cases of not using `F` if it's safe. I would classify these cases as acceptable, yes :)

I'd like a comment though, because someone reading this will wonder why it was done this way and not the "proper" way.

Otoh, in this case we already have a `Sync` bound so we could `repeatEval`, but still, a comment is enough for me